receiver/azuremonitorreceiver/scraper.go

```diff
@@ -241,6 +238,20 @@ func (s *azureScraper) getSubscriptions(ctx context.Context) {
 		return
 	}
 
+	if !s.cfg.DiscoverSubscriptions {
+		// if there's only subscription ids configured, we need to get additional info
+		for _, subID := range s.cfg.SubscriptionIDs {
+			resp, err := armSubscriptionClient.Get(ctx, subID, &armsubscriptions.ClientGetOptions{})
+			if err != nil {
```

Comment:
Should we stop trying to get the rest of the subscriptions because one in the middle of the list failed?

Reply:
If it does fail, should we be calling `unloadSubscription`?

Reply:
Is there also any risk of this being a permissions issue, and we shouldn't disable the rest of collection because we couldn't get the name?

Reply:
I decided to break if the sub get does not work because this is the sub id list provided by the user in config.
If one sub id is wrong, one should fix the config. That was the spirit. 

Now that's true that if someone doesn't want the subscription name, it's kinda useless to make these get. 

You make me think of something that I didn't think at all from the beginning that I got back the project: the client configuration in Azure. I will check twice how we configured it and make a doc explaining that part. Indeed here that would need probably an additional scope or something.

Thanks for the light!